pkgs/by-name/cc/ccache/package.nix

```
@@ -26,7 +26,20 @@ stdenv.mkDerivation (finalAttrs: {
     owner = "ccache";
     repo = "ccache";
     rev = "refs/tags/v${finalAttrs.version}";
-    sha256 = "sha256-0T9iJXnDX8LffhB/5hsfBNyZQ211f0lL/7dvTrjmiE0=";
+    # `git archive` replaces `$Format:%H %D$` in cmake/CcacheVersion.cmake
+    # we need to replace it with something reproducible
+    # see https://github.com/NixOS/nixpkgs/pull/316524
+    postFetch = ''
+      sed -i -E \
+        's/version_info "([0-9a-f]{40}) .*(tag: v[^,]+)"/version_info "\1 \2"/g w match' \
```

Thanks for looking into this in more depth @dotlambda!! I didn't realize git archives can be made non-reproducible like this. I found [one other case of this in nixpkgs](https://github.com/dotlambda/nixpkgs/blob/a37a6e107f4bf62cfec9202818468b4b4a918c26/pkgs/applications/science/misc/snakemake/default.nix#L17-L20) which lead me to https://github.com/python-versioneer/python-versioneer/issues/217.

It looks like we should also extend the match to include anything after the tag, and before the closing quote:

```suggestion
        's/version_info "([0-9a-f]{40}) .*(tag: v[^,]+).*"/version_info "\1 \2"/g w match' \
```